### DISCUSSION TITLE
AaC Contribution Workflow

### DISCUSSION THREAD

**Discussion by jondavid-black** [DevOps-MBSE/AaC#50] [Ideas]
I'm finding myself adjusting my workflow a bit to get more value from the GitHub Actions CI/CD we have in place.  I'm creating my PR earlier so I can push and see that the pipeline works but not really intending the PR to be merged at that time.  I think this could create some confusion...is a PR ready or just there for convenience?

I really like being able to push more frequently (because my local repo could die with the next patch cycle).

So I'm thinking we should create a new column in the project called ready.  We can add a PR to that column when it is ready for merge to main...adding a step to our workflow but overcoming potential confusion.

I'm going to go ahead and create the column, but I'd appreciate your feedback and we can adjust as needed.  What do you think about this adjusted workflow?  Are there other ideas on how we could improve?

**Comment by Coffee2Bits** [comment]
Would adding the Github Actions to all branches address this issue? 

I would have suggested it already, but I do want to be conscientious about the Github Actions minutes. We should have about 2k minutes per month for [Github Actions](https://docs.github.com/en/billing/managing-billing-for-github-actions/about-billing-for-github-actions#included-storage-and-minutes), and it looks like the pipeline is about 2 minutes long. So, if we keep it to less than about 30 jobs per day we should be fine for Github Actions.  
```(2000 minutes per month / 2 min pipeline / 30 days in a month = 33 pipeline runs per Day @ 2 min per pipeline)```

**Comment by cvchaparro** [comment]
I've been trying to push fairly frequently so I can get the pipeline feedback (although I test locally, too, before pushing usually). I think having a Draft PR has worked for me to represent a PR that's not ready to be merged, yet, but that lets us have the pipeline feedback.

**Comment by jondavid-black** [comment]
I see that you've been using the draft PR.  I've never used that feature before.  Let's give that a try because it does make things pretty clear and simple.

**Comment by lizzcondrey** [comment]
So, the ready for merge column is in place and in used currently by the team. The draft PR portion is also quite often adhered to. And it has seemed to be practice that reviewers are not added to the PR until it is ready for actual review. With those three pieces, I think this discussion is eligible for closure. Thoughts?

**Comment by lizzcondrey** [comment]
Per the confirmation reaction above, closing out this discussion page.
